conda_smithy/feedstock_tokens.py

```diff
@@ -175,6 +185,75 @@ def feedstock_token_exists(user, project, token_repo):
     return exists
 
 
+def is_valid_feedstock_token(user, project, feedstock_token, token_repo):
+    """Test if the input feedstock_token is valid.
+
+    All exceptions are swallowed and stdout/stderr from this function is
+    redirected to `/dev/null`. Sanitized error messages are
+    displayed at the end.
+
+    If you need to debug this function, define `DEBUG_FEEDSTOCK_TOKENS` in
+    your environment before calling this function.
+    """
+    from .github import gh_token
+
+    github_token = gh_token()
+
+    failed = False
+    err_msg = None
+    valid = False
+
+    # capture stdout, stderr and suppress all exceptions so we don't
+    # spill tokens
+    with tempfile.TemporaryDirectory() as tmpdir, open(
+        os.devnull, "w"
+    ) as fp, redirect_stdout(fp), redirect_stderr(fp):
+        try:
+            # clone the repo
+            _token_repo = (
+                token_repo.replace("$GITHUB_TOKEN", github_token)
+                .replace("${GITHUB_TOKEN}", github_token)
+                .replace("$GH_TOKEN", github_token)
+                .replace("${GH_TOKEN}", github_token)
```

Comment:
What's this supposed to do?

Reply:
The feedstock token repo is private, so to pull the repo I put the token in the web address. This allows the user to input a web address with `${GITHUB_TOKEN}` etc in it and then the pull will work.